### PR TITLE
Add Vanguard65 Keyboard

### PR DESCRIPTION
**Pull request by mrnoisytiger**
## Description

Made some changes to the Keebfront Vanguard65 to better comply with QMK best practices. Thanks!

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/keebfront/vanguard65/keyboard.json [hunk 1]
```diff
@@ -0,0 +1,743 @@
...
+    "manufacturer": "Keebfront",
+    "keyboard_name": "Vanguard65",
+    "maintainer": "Keebfront",
+    "bootloader": "atmel-dfu",
+    "build": {
+        "lto": true
+    },
+    "diode_direction": "COL2ROW",
+    "encoder": {
+        "enabled": true,
```
Comment (waffle87) [inline keyboards/keebfront/vanguard65/keyboard.json:11]:
```suggestion
```

File: keyboards/keebfront/vanguard65/keyboard.json [hunk 2]
```diff
@@ -0,0 +1,743 @@
...
+        "rotary": [
+            {"pin_a": "E6", "pin_b": "D4"}
+        ]
+    },
+    "features": {
+        "extrakey": true,
+        "midi": true,
+        "rgblight": true
```
Comment (waffle87) [inline keyboards/keebfront/vanguard65/keyboard.json:19]:
```suggestion
        "rgblight": true,
        "mousekey": true,
        "encoder": true
```

File: keyboards/keebfront/vanguard65/keyboard.json [hunk 3]
```diff
@@ -0,0 +1,743 @@
...
+    "processor": "atmega32u4",
+    "rgblight": {
+        "animations": {
+            "alternating": true,
+            "breathing": true,
+            "christmas": true,
+            "knight": true,
+            "rainbow_mood": true,
+            "rainbow_swirl": true,
+            "rgb_test": true,
```
Comment (waffle87) [inline keyboards/keebfront/vanguard65/keyboard.json:34]:
```suggestion
```
(Or maybe remove more of these if you're tight on space)

File: keyboards/keebfront/vanguard65/config.h [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+You should have received a copy of the GNU General Public License
+along with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+/* Define MIDI */
+#define MIDI_ADVANCED
+#define SLIDER_PINA D6
```
Comment (waffle87) [inline keyboards/keebfront/vanguard65/config.h:22]: This does not appear to be used anywhere?

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/keebfront/vanguard65/config.h (hunk 4 above)
Comment (mrnoisytiger) [reply keyboards/keebfront/vanguard65/config.h:22]: @waffle87 This is actually used extensively in the MIDI definition, but the features are not present in the `default` keymap as I understand QMK wants to have it be clean. 

Is there a different way to define this so the lines aren't in QMKs repo but still will activate when compiling for VIA?

Comment (zvecr) [reply keyboards/keebfront/vanguard65/config.h:22]: If there is a slider on the board, the expectation would be for code to exist to control its behaviour.

"clean" doesnt mean zero code in the default keymap.

Comment (mrnoisytiger) [reply keyboards/keebfront/vanguard65/config.h:22]: Oh I see the confusion. Last time I understood it to mean "nothing besides a keymap" in there. 

I'll add the basic MIDI functionality in (which in my view is the core), but there is a huge codebase involved for the fully-fledged functionality which is inseparable from VIA's web UI. Guessing you guys don't want all of that code-base though.

**Review by waffle87 — COMMENTED** [review]
I've marked the above suggestion as unresolved. What was trying to be conveyed earlier is that if this hardware (a potentiometer presumably) is intended to be installed on the keyboard, there should be code to support in the firmware.

In it's current state, I see this firmware (having the `#define SLIDER_PINA D6`, but no code to utilise it) as incomplete.

**Comment by mrnoisytiger** [comment]
Kindly requesting re-review since this version should meet all the guidelines! Thanks!

File: keyboards/keebfront/vanguard65/readme.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
+# Keebfront Vanguard65
+
+![Vanguard65](https://i.imgur.com/1AnyRaEh.png)
+The following is the QMK Firmware for the Keebfront Vanguard65
+
+
+* Keyboard Maintainer: Keebfront
+* Hardware Supported: Vanguard65
+
```
Comment (drashna) [inline keyboards/keebfront/vanguard65/readme.md:9]: No link for where to get or have gotten?

File: keyboards/keebfront/vanguard65/keymaps/default/rules.mk [hunk 6]
```diff
@@ -0,0 +1,2 @@
+ENCODER_MAP_ENABLE = yes
+SRC += analog.c
```
Comment (zvecr) [inline keyboards/keebfront/vanguard65/keymaps/default/rules.mk:2]:
```suggestion
ANALOG_DRIVER_REQUIRED = yes

```

File: keyboards/keebfront/vanguard65/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+                 KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                    KC_RALT, MO(1),            KC_LEFT, KC_DOWN, KC_RGHT
+    ),
+
+    [_LAYER1] = LAYOUT_ansi_blocker_split_bs(
+        _______, QK_GESC, KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  _______, _______, KC_END,
+                 _______, RGB_TOG, RGB_MOD, RGB_HUI, RGB_HUD, RGB_SAI, RGB_SAD, RGB_VAI, RGB_VAD, _______, KC_PSCR, KC_SCRL, KC_PAUS, QK_BOOT,          _______,
+                 _______, RGB_SPI, RGB_SPD, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,          _______,
+                 _______,          _______, _______, _______, _______, _______, NK_TOGG, _______, _______, _______, _______, _______, KC_VOLU,
+                 _______, _______, _______,                            _______,                   _______, _______,          KC_MPRV, KC_VOLD, KC_MNXT
+    ),
```
Comment (zvecr) [inline keyboards/keebfront/vanguard65/keymaps/default/keymap.c:44]:
```suggestion
    [_LAYER1] = LAYOUT_all(
        _______, QK_GESC, KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  _______, _______, KC_END,
                 _______, RGB_TOG, RGB_MOD, RGB_HUI, RGB_HUD, RGB_SAI, RGB_SAD, RGB_VAI, RGB_VAD, _______, KC_PSCR, KC_SCRL, KC_PAUS, QK_BOOT,          _______,
                 _______, RGB_SPI, RGB_SPD, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,          _______,
                 _______, _______, _______, _______, _______, _______, _______, NK_TOGG, _______, _______, _______, _______, _______, KC_VOLU,
                 _______, _______, _______,                            _______,                   _______, _______,          KC_MPRV, KC_VOLD, KC_MNXT
    )
```

File: keyboards/keebfront/vanguard65/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU General Public License for more details.
+You should have received a copy of the GNU General Public License
+along with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include QMK_KEYBOARD_H
+#include "analog.h"
+#include "string.h"
+
```
Comment (zvecr) [inline keyboards/keebfront/vanguard65/keymaps/default/keymap.c:18]: This shouldn't be required
```suggestion

```

File: keyboards/keebfront/vanguard65/keymaps/default/keymap.c [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+uint8_t last_val = 0;
+uint8_t current_val = 0;
```
Comment (zvecr) [inline keyboards/keebfront/vanguard65/keymaps/default/keymap.c:20]:
```suggestion
```

File: keyboards/keebfront/vanguard65/keymaps/default/keymap.c [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+
+#if defined(ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
+    [_LAYER0] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU) },
+    [_LAYER1] = { ENCODER_CCW_CW(KC_TRNS, KC_TRNS) },
+};
+#endif
+
+void slider(void) {
+    current_val = analogReadPin(SLIDER_PINA) >>3;
```
Comment (zvecr) [inline keyboards/keebfront/vanguard65/keymaps/default/keymap.c:55]:
```suggestion
    uint8_t current_val = analogReadPin(SLIDER_PINA) >>3;
```

**Review by zvecr — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/keebfront/vanguard65/keyboard.json [hunk 11]
```diff
@@ -0,0 +1,743 @@
...
+    "build": {
+        "lto": true
+    },
+    "diode_direction": "COL2ROW",
+    "encoder": {
+        "rotary": [
+            {"pin_a": "E6", "pin_b": "D4"}
+        ]
+    },
+    "features": {
```
Comment (waffle87) [inline keyboards/keebfront/vanguard65/keyboard.json:15]:
```suggestion
    "features": {
        "bootmagic": true,
```

Comment (mrnoisytiger) [reply keyboards/keebfront/vanguard65/keyboard.json:15]: Interesting, did Bootmagic get stripped from the default behavior? I always thought it was included as default.

**Review by waffle87 — APPROVED** [review]

File: keyboards/keebfront/vanguard65/keyboard.json (hunk 11 above)
Comment (drashna) [reply keyboards/keebfront/vanguard65/keyboard.json:15]: Via does enable it by default, but otherwise, no it's not enabled.

Comment (mrnoisytiger) [reply keyboards/keebfront/vanguard65/keyboard.json:15]: Good to know! Thank you!

**Pull request merged**